### ISSUE TITLE
Complains about duplicate variable even though I use let instead of var

### ISSUE THREAD

**Issue by wereHamster** [palantir/tslint#295]
The following code generates a `duplicate variable: something` warning even though the code is perfectly fine.

```javascript
if (condition) {
    let something = 1;
} else {
    let something = 2;
}
```

**Comment by wereHamster** [comment]
Awesome. Thanks.
